nebula-logger/plugins/log-retention-rules/plugin/classes/LogRetentionFilter.cls

```diff
@@ -0,0 +1,464 @@
+                    return this.compareString((String) recordFieldValue, String.valueOf(comparisonValue));
+                }
+                when else {
+                    throw new IllegalArgumentException('Could not process field path: ' + this.filterCondition.FieldPath__c);
+                }
+            }
+        }
+
+        private Boolean matchesRegEx() {
+            Pattern pattern = Pattern.compile(this.filterCondition.Value__c);
+            return pattern.matcher(String.valueOf(this.getFieldValue())).matches();
+        }
+
+        private Object getFieldValue() {
+            return new FieldPath(this.sobjectType, this.filterCondition.FieldPath__c).getValue(this.record);
+        }
+
+        private Object getComparisonValue() {
+            switch on this.filterCondition.ValueType__c {
+                when 'Field' {
+                    return new FieldPath(this.sobjectType, this.filterCondition.Value__c).getValue(this.record);
+                }
+                when 'RegEx' {
+                    return this.filterCondition.Value__c;
+                }
+                when 'Value' {
+                    return this.filterCondition.Value__c;
+                }
+                when else {
+                    throw new IllegalArgumentException('Unknown Value Type, cannot parse comparison value');
+                }
+            }
+        }
+
+        // Helper methods for dealing with converting field values & strings
+        // (stored in CMDT) to the appropriate data type
+        private Datetime getAsDatetime(Object datetimeValue) {
+            if (datetimeValue == null) {
+                return null;
+            } else if (datetimeValue instanceof Datetime) {
+                return (Datetime) datetimeValue;
+            } else {
+                String datetimeString = (String) datetimeValue;
+                return (Datetime) JSON.deserialize(datetimeString, Datetime.class);
+            }
+        }
+
+        private Decimal getAsDecimal(Object decimalValue) {
+            if (decimalValue == null) {
+                return null;
+            } else if (decimalValue instanceof Decimal) {
+                return (Decimal) decimalValue;
+            } else {
+                String decimalString = (String) decimalValue;
+                return (Decimal) JSON.deserialize(decimalString, Decimal.class);
+            }
+        }
+
+        // In Apex, you can't use comparison operators on instances of Object, so several private methods are used for each data type
+        // Example of what you can't do in Apex:
+        //          Object today = System.today();
+        //          Object yesterday = System.today().addDays(-1);
+        //          System.assert(today > yesterday); // This line cannot execute since it's comparing Object
+        private Boolean compareDatetime(Datetime recordFieldValue, Datetime comparisonValue) {
+            switch on this.filterCondition.Operator__c {
```

Comment:
same thing here, with Operator__c. Again, no expectations as far as making these changes as part of this PR, but good food for thought!

Reply:
Same, I'll add a `TODO` for now, but I love the idea of using a private enum